demo/src/test/java/uk/co/ribot/easyadapterdemo/EasyRecyclerAdapterTest.java

```diff
@@ -0,0 +1,181 @@
+        assertEquals(0, mEasyRecyclerAdapter.getItems().size());
+        mEasyRecyclerAdapter.setItemsWithoutNotifying(list);
+        assertEquals(list, mEasyRecyclerAdapter.getItems());
+        verify(mMockAdapterDataObserver, never()).onChanged();
+    }
+
+    @Test public void testGetItems() throws Exception {
+        List<Person> list = getSomePeople();
+        EasyRecyclerAdapter<Person> easyRecyclerAdapter = new EasyRecyclerAdapter<>(
+                RuntimeEnvironment.application,
+                PersonViewHolder.class,
+                list);
+        assertEquals(list, easyRecyclerAdapter.getItems());
+    }
+
+    @Test public void testAddItem() throws Exception {
+        mEasyRecyclerAdapter.getItems().addAll(getSomePeople());
+        Person newPerson = createPerson("New Person");
+        mEasyRecyclerAdapter.addItem(newPerson);
+
+        int indexOfItem = mEasyRecyclerAdapter.getItems().indexOf(newPerson);
+        assertTrue(indexOfItem > 0);
+        verify(mMockAdapterDataObserver).onItemRangeInserted(indexOfItem, 1);
+    }
+
+    @Test public void testRemoveItem() throws Exception {
+        List<Person> items = getSomePeople();
+        Person personToRemove = items.get(0);
+        mEasyRecyclerAdapter.getItems().addAll(items);
+        boolean result = mEasyRecyclerAdapter.removeItem(personToRemove);
+
+        assertTrue(result);
+        assertFalse(mEasyRecyclerAdapter.getItems().contains(personToRemove));
+        verify(mMockAdapterDataObserver).onItemRangeRemoved(0, 1);
+    }
+
+    @Test public void testRemoveNonExistingItem() throws Exception {
+        Person person = createPerson("Person1");
+        boolean result = mEasyRecyclerAdapter.removeItem(person);
+
+        assertFalse(result);
+        verify(mMockAdapterDataObserver, never()).onChanged();
+        verify(mMockAdapterDataObserver, never()).onItemRangeRemoved(anyInt(), anyInt());
+    }
+
+    @Test public void testAddItems() throws Exception {
+        Person person1 = createPerson("Person1");
+        Person person2 = createPerson("Person2");
+        Person person3 = createPerson("Person3");
+        mEasyRecyclerAdapter.getItems().add(person1);
+        List<Person> listToAdd = Arrays.asList(person2, person3);
+        boolean result = mEasyRecyclerAdapter.addItems(listToAdd);
+
+        assertTrue(result);
+        assertEquals(3, mEasyRecyclerAdapter.getItemCount());
+        assertTrue(mEasyRecyclerAdapter.getItems().contains(person2));
+        assertTrue(mEasyRecyclerAdapter.getItems().contains(person3));
+        verify(mMockAdapterDataObserver).onItemRangeInserted(1, listToAdd.size());
+    }
+
+    @Test public void testRemoveItems() throws Exception {
+        Person person1 = createPerson("Person1");
+        Person person2 = createPerson("Person2");
+        Person person3 = createPerson("Person3");
+        mEasyRecyclerAdapter.getItems().addAll(Arrays.asList(person1, person2, person3));
```

Comment:
You know what I'm gonna say  
![pny](https://cloud.githubusercontent.com/assets/3879281/8059943/66a89592-0eba-11e5-85ae-ebe66ab49cc7.gif)